tower-balance/src/load/mod.rs

```diff
@@ -0,0 +1,52 @@
+use futures::Poll;
+use tower::Service;
+
+mod constant;
+mod pending_requests;
+
+pub use self::constant::{Constant, WithConstant};
+pub use self::pending_requests::{PendingRequests, WithPendingRequests};
+
+/// A `Service` that has a load metric when it is ready.
+pub trait PollLoad: Service {
+    /// Returns a `Load` metric for the underlying `Service`, if the service is ready.
+    ///
+    /// Must not return a value unless `Service::poll_ready` would return `Async::Ready`.
+    fn poll_load(&mut self) -> Poll<Load, Self::Error>;
```

Comment:
Alternatively, this could just be `fn load(&self) -> Load` and say that it's fatal to call it when `poll_ready` is not ready.

Reply:
Yeah, you're right that load can be called when not ready...